src/Screens/ConfirmationScreen/confirmationPage.tsx

```diff
@@ -1,10 +1,233 @@
+import Image from "react-native-elements/dist/image/Image";
+import uuid from "uuid";
+
+import DropDownPicker from "react-native-dropdown-picker";
+import { confirmationPageStyles } from "../../styles/confirmationPageStyles";
+import { reserveTableScreenStyles } from "../../styles/reserveTableScreenStyles";
+import { formatAdress } from "../../utils/adressUtils";
+
+const CARDS = [
+  { label: "card1", value: "card1" },
+  { label: "card2", value: "card2" },
+  { label: "card3", value: "card3" },
+  { label: "Новая карта", value: "Привязать новую карту" },
+];
+
+const confirmationScreen = ({ cart, totalSum, route }) => {
+  const [open, setOpen] = useState(false);
+  const [value, setValue] = useState("");
+  const [orderData, setOrderData] = useState(route.params);
+  const [excludedIngredients, setExcludedIngredients] = useState([]);
+
+  useEffect(() => {
+    const excludedIng = cart.map((cartElement) => {
+      const selectedIngredients = cartElement.ingredients.split(";");
+      const allIngredients = cartElement.dish.ingredients.split(";");
+      const excludedIngredients = allIngredients.filter(
+        (ingredient) => !selectedIngredients.includes(ingredient)
+      );
+      return excludedIngredients;
+    });
+
+    setExcludedIngredients(excludedIng);
+  }, [cart]);
 
-export const confirmationScreen = () => {
   return (
-    <View>
-      <Text>HELLO FROM CONFIRMATION PAGE</Text>
+    <View style={confirmationPageStyles.mainWrapper}>
+      <ScrollView
+        style={confirmationPageStyles.scrollViewStyle}
+        nestedScrollEnabled={true}
+      >
+        <View>
+          {cart.map((item, index) => (
+            <View style={confirmationPageStyles.mainContainer} key={uuid.v4()}>
+              <View style={confirmationPageStyles.imageWithCountStyles}>
+                <Image
+                  source={{
+                    uri: item.dish.imageURL,
+                  }}
+                  style={confirmationPageStyles.imageStyle}
+                />
+                <View style={confirmationPageStyles.quantityContainer}>
+                  <Text style={confirmationPageStyles.quantityItem}>
+                    X{item.quantity}
+                  </Text>
+                </View>
+              </View>
+              <View style={confirmationPageStyles.textItemsContainer}>
+                <View>
+                  <Text style={confirmationPageStyles.itemTitle}>
+                    {item.dish.name}
+                  </Text>
+                </View>
+                {excludedIngredients[index]?.length > 0 ? (
+                  <View style={confirmationPageStyles.excludedIngWrapper}>
+                    <Text style={confirmationPageStyles.withoutIngText}>
+                      Без добавления:
+                    </Text>
+                    <Text style={confirmationPageStyles.ingTextWrapper}>
+                      {excludedIngredients[index]?.join(", ")}
+                    </Text>
+                  </View>
```

Comment:
move to the common component and reuse after

Reply:
fixed